api/v1/storagecluster_types.go

```diff
@@ -137,11 +140,42 @@ type ManageCephObjectStoreUsers struct {
 	ReconcileStrategy string `json:"reconcileStrategy,omitempty"`
 }
 
+// ExternalStorageKind specifies a kind of the external storage
+type ExternalStorageKind string
+
+const (
+	// KindOcs specifies a "ocs" kind of the external storage
+	KindOcs ExternalStorageKind = "ocs"
+
+	// KindRhcs specifies a "rhcs" kind of the external storage
+	KindRhcs ExternalStorageKind = "rhcs"
+)
+
 // ExternalStorageClusterSpec defines the spec of the external Storage Cluster
-// to be connected to the local cluster
+// to be connected to the external cluster
 type ExternalStorageClusterSpec struct {
-	// +optional
+	//+kubebuilder:default:=true
+	// Enable enables the external mode
 	Enable bool `json:"enable,omitempty"`
+
+	//+kubebuilder:default:=rhcs
+	// StorageProviderKind describes the kind of the external storage
+	StorageProviderKind ExternalStorageKind `json:"storageProviderKind,omitempty"`
+
+	// ConnectionString describes the endpoint and UID of the provider
```

Comment:
should it be a secret name instead?

Reply:
+1 for the secret

Reply:
@sp98 @iamniting  Why complicate stuff with a secret? It will force the customer to create a secret for non-confidential information? will complicate the setup and generally speaking does not make sense in my eyes.

Reply:
Ok, it was decided in an offline conversation to go with an encoded string (base64?) provided at the `Spec.ExternalStorage.ExternalStorageKind` field

Reply:
correct and made the changes also to the API comment